### PR TITLE
Feature/27/integration test for es integration

### PR DESCRIPTION
**Pull request by YMikevich**

### CODE REVIEW

File: clients/es-integration/build.gradle [hunk 1]
```diff
@@ -12,13 +13,45 @@ checkstyle {
     toolVersion "6.7"
 }
 
+idea {
```
Comment (akraskovski) [inline clients/es-integration/build.gradle]: why did you add this settings section?

File: clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/EsIntegrationApplicationContextIT.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.github.ymikevich.es.integration;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class EsIntegrationApplicationContextIT {
+
+    public static void main(final String[] args) {
+        SpringApplication.run(EsIntegrationApplication.class, args);
```
Comment (akraskovski) [inline clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/EsIntegrationApplicationContextIT.java]: Are you sure, that you're running a valid spring application here?

File: clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/services/DefaultTweetServiceIT.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = EsIntegrationApplicationContextIT.class)
```
Comment (akraskovski) [inline clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/services/DefaultTweetServiceIT.java]: do you need to duplicate configurations context here?

File: clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/services/DefaultTweetServiceIT.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+public class DefaultTweetServiceIT {
+
+    @Autowired
+    WebApplicationContext context;
+
+    @Autowired
+    TweetRepository repository;
+
+    @Autowired
+    TweetController controller;
```
Comment (akraskovski) [inline clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/services/DefaultTweetServiceIT.java]: if you're writing mvc test - all communications with rest layer should be done via MockMvc instance and not of directly acquiring controllers

File: clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/services/DefaultTweetServiceIT.java [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+            + "}";
+
+
+    @Before
+    public void setUp() {
+        mockMvc = MockMvcBuilders.webAppContextSetup(context)
+                .build();
+    }
+
+    private Tweet generateTweet(final TwitterUser user, final Long id, final int favoriteCount,
```
Comment (akraskovski) [inline clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/services/DefaultTweetServiceIT.java]: where private method must be placed?

File: clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/services/DefaultTweetServiceIT.java [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+            + "    \"retweetValue\": 500,\n"
+            + "    \"tweetId\": \"375685667\",\n"
+            + "    \"source\": \"http://somesource.com\"\n"
+            + "  }\n"
+            + "}";
+
+
+    @Before
+    public void setUp() {
+        mockMvc = MockMvcBuilders.webAppContextSetup(context)
```
Comment (akraskovski) [inline clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/services/DefaultTweetServiceIT.java]: I think you're doing the same job here as spring could do for you by configuring MockMvc with standard context

File: clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/services/DefaultTweetServiceIT.java [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+    @Test
+    public void searchTest() throws Exception {
+        var user = new TwitterUser();
+        user.setId("10");
+        user.setEmail("bred@mail.com");
+        user.setName("humortranslate");
+        var tweet1 = generateTweet(user, 125324235L, 250, 300, "Hey");
+        var tweet2 = generateTweet(user, 265473423L, 100, 150, "Hello Hey");
+        var tweet3 = generateTweet(user, 375685667L, 500, 1000, "Bye");
+        var tweets = new ArrayList<Tweet>();
```
Comment (akraskovski) [inline clients/es-integration/src/integration-test/java/com/github/ymikevich/es/integration/services/DefaultTweetServiceIT.java]: List.of() was provided as a good factory method in java 11

File: clients/es-integration/src/integration-test/resources/bootstrap.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+spring:
+  rabbitmq:
```
Comment (akraskovski) [inline clients/es-integration/src/integration-test/resources/bootstrap.yml]: rabbit in integration tests?!

File: clients/es-integration/src/integration-test/resources/bootstrap.yml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    host: ${RABBIT_HOST:localhost}
+    username: ${RABBIT_USERNAME:guest}
+    password: ${RABBIT_PASSWORD:guest}
+  data:
+    elasticsearch:
+      cluster-nodes: ${ELASTICSEARCH_CLUSTER_NODES:localhost:9300}
+      cluster-name: ${ELASTICSEARCH_CLUSTER_NAME:tweets-cluster}
+  cloud:
```
Comment (akraskovski) [inline clients/es-integration/src/integration-test/resources/bootstrap.yml:6]: cloud in integration tests?!

File: es-integration-it.env [hunk 10]
```diff
@@ -0,0 +1,7 @@
+RABBIT_HOST=rabbitmq
```
Comment (akraskovski) [inline es-integration-it.env]: check, please

File: es-integration-it.yml [hunk 11]
```diff
@@ -0,0 +1,41 @@
+version: '3'
+
+services:
+  rabbitmq:
```
Comment (akraskovski) [inline es-integration-it.yml]: ????

File: es-integration-it.yml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+      - spring-cloud-services-network
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch:6.4.2
+    container_name: elasticsearch
+    environment:
+      - node.name=elasticsearch
+      - discovery.type=single-node
+      - cluster.name=tweets-cluster
+    volumes:
+      - 'esdata0:/usr/share/elasticsearch/data'
```
Comment (akraskovski) [inline es-integration-it.yml]: there is no need to share volume to a host machine during tests execution

File: es-integration-it.yml [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+    depends_on:
+      - rabbitmq
+      - elasticsearch
+    env_file:
+      - es-integration-it.env
+    build:
+      context: ./clients/es-integration
+      dockerfile: IntegrationTestDockerfile
+    volumes:
+      - '$HOME/tweets-analyzer/es-integration/logs:/clients/es-integration/logs'
```
Comment (akraskovski) [inline es-integration-it.yml]: also here

**Review by akraskovski — CHANGES_REQUESTED** [review]

File: clients/es-integration/build.gradle (hunk 1 above)
Comment (YMikevich) [reply clients/es-integration/build.gradle]: Added to separate idea and gradle output directories

File: clients/es-integration/src/integration-test/resources/bootstrap.yml (hunk 9 above)
Comment (YMikevich) [reply clients/es-integration/src/integration-test/resources/bootstrap.yml:6]: This configuration disables config server and service discovery
